### DISCUSSION TITLE
Allow passing a componet to DonutChart > chartLabel prop

### DISCUSSION THREAD

**Discussion by RenanStigliani** [mantinedev/mantine#6917] [Feature requests]
Could we change [DonutChart](https://mantine.dev/charts/donut-chart/) > `chartLabel` props to accept a component? I want to use a <Stack> to achieve something like the below:

![image](https://github.com/user-attachments/assets/2d76bfa0-1f24-4026-aafb-aa363b091c0b)

Thanks for the awesome library!

**Comment by rtivital** [comment]
Chart label is rendered within svg element, it is not possible to render any html content there. If you need this feature, use RingProgress component instead.

**Reply by RenanStigliani** [reply]
Looking at [this code](https://github.com/mantinedev/mantine/blob/397e782958d44d66afb3fbdda53860e599682306/packages/%40mantine/charts/src/DonutChart/DonutChart.tsx#L230), doesn't look like it's inside an svg (I may be overlooking something thou).

I assumed we could just render is instead of the text element there

**Reply by rtivital** [reply]
Nope, recharts has strict children filtering, only svg children are allowed
